### ISSUE TITLE
SC2128 falselly applies in if/then clauses

### ISSUE THREAD

**Issue by nkakouros** [koalaman/shellcheck#1043]
#### For bugs
- Rule Id: SC2128
- My shellcheck version (`shellcheck --version` or "online"): 0.4.6
- [X] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
if [ "$world" == "all" ]; then
  worlds=$(grep -E "^*: enabled" $PROJECT_DIR/vars/worlds.yml | cut -d: -f1)
  IFS=$'\n' readarray worlds <<< $worlds
elif [ -n "$world" ]; then
  IFS="," read -r -a worlds <<< "$worlds" 
fi
```

#### Here's what shellcheck currently says:
```
Line 5:
$worlds
^-- SC2128: Expanding an array without an index only gives the first element.
```

#### Here's what I wanted or expected to see:
No error (unless I am understanding the code snippet wrongly).

**Comment by koalaman** [comment]
You're right, ShellCheck is just confused by the fact that you use the name `worlds` in turn for both a string and an array.  As a workaround, you can use different variables.
